### PR TITLE
Integration of go.uber.org/zap logger

### PR DESCRIPTION
**Pull request by sshumakov**
Replaced logger interface and default logger with go.uber.org/zap logger. Zap logger is structured logger where you can pass params as key-value pairs.

### CODE REVIEW

File: network/config.go [hunk 1]
```diff
@@ -38,9 +39,11 @@ var DefaultConfig = Config{
 	discovery:  time.Second * 2,
 }
 
+var defaultLogger, _ = zap.NewDevelopment()
```
Comment (awishformore) [inline network/config.go]: Should still be a public variable, so people are free to use it explicitly if they want.

**Comment by sshumakov** [comment]
Checked logging, added example. Zap doesn't send structured properties however, created issue inside zap repository - https://github.com/uber-go/zap/issues/418.

File: network/config.go (hunk 1 above)
Comment (sshumakov) [inline network/config.go]: Agree, makes sense

**Comment by sshumakov** [comment]
Okay, finished with code review changes, @awishformore please review once you have time.

File: example/many/many.go [hunk 2]
```diff
@@ -37,8 +37,26 @@ func main() {
...
+	// logLevelMap[zapcore.WarnLevel] = appinsights.Warning
+	// logLevelMap[zapcore.ErrorLevel] = appinsights.Error
+	// logLevelMap[zapcore.DPanicLevel] = appinsights.Critical
+	// logLevelMap[zapcore.PanicLevel] = appinsights.Critical
+	// logLevelMap[zapcore.FatalLevel] = appinsights.Critical
+
+	// log, _ := zap.NewDevelopment(zap.Hooks(func(entry zapcore.Entry) error {
+	// 	client.TrackTraceTelemetry(appinsights.NewTraceTelemetry(entry.Message, logLevelMap[entry.Level]))
+	// 	return nil
+	// }))
```
Comment (awishformore) [inline example/many/many.go]: This seems cool, but it belongs into a document file (logger.md in a logs folder with this as example code would be nice).

**Comment by awishformore** [comment]
I think the fact that it's really hard to create an interface when trying to use a structured logger means this is probably the way to go for now. Especially if we consider managing big infrastructures with many Veltor nodes, this should be very useful. Good job :+1:

**Comment by sshumakov** [comment]
Okay, moved hook example to logger.md file

**Comment by awishformore** [comment]
Merged manually in #26 to fix LRCF in `logger.md`, add new line to `AUTHORS` file and make some logging messages more concise.

**Pull request closed without merging**